routes/helpers/localize_results.js

```diff
@@ -0,0 +1,37 @@
+const localize =  (locale, dbResultsObj) => {
+
+  // keynames in DB that contain translatable content follow pattern 'name___locale'
+  const separator = '___';
+  const translatableKeyReg = /\_{3}.+$/;
+
+  // gets array of translatable values from DB results
+  const translatableFields = Object.keys(dbResultsObj)
+    .filter(key => translatableKeyReg.test(key))
+    .map(filteredKey => filteredKey.replace(translatableKeyReg, ''));
+
+  // create new results object with relevant data only
+  return Object.keys(dbResultsObj)
+    .filter(key => {
+      const [ type, loc, ] = key.split(separator);
+      return translatableFields.indexOf(type) < 0 || loc === locale;
```

Comment:
Yeaaah you're kind of right. Took me a while to figure it out. I'm glad I wrote tests. I still think it needs 2 conditions, however the first condition is less complicated now.